src/lib.rs

```diff
@@ -118,3 +132,89 @@ impl Relation {
         };
     }
 }
+
+pub enum JfRelation {
+    Withdraw,
+}
+
+impl JfRelation {
+    pub fn generate_keys(&self) -> (JfProvingKey<Curve>, JfVerifyingKey<Curve>) {
+        let rng = &mut jf_utils::test_rng();
+        let srs = generate_srs(17_000, rng).unwrap();
```

Comment:
this shouldn't be benchmarked

Reply:
Rand generators are usually initialized once and getting a handle is almost free, it takes nanoseconds. Good point with srs. I've noticed also that for many operations, we generate circuit from scratch, e.g. for proof generation.

Reply:
Fixed.